### PR TITLE
Remove LegacyESVersion.V_6_4_x constants

### PR DESCRIPTION
**Pull request by adnapibar**
### Description
This PR removes LegacyESVersion.V_6_4_x constants. 

### Issues Resolved
Part of the meta issue #1674 
 
### Check List
- [x] All tests pass
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

Signed-off-by: Rabi Panda <adnapibar@gmail.com>

### CODE REVIEW

File: server/src/main/java/org/opensearch/search/slice/SliceBuilder.java [hunk 1]
```diff
@@ -227,20 +227,17 @@ public Query toFilter(ClusterService clusterService, ShardSearchRequest request,
 
         int shardId = request.shardId().id();
         int numShards = context.getIndexSettings().getNumberOfShards();
-        if (minNodeVersion.onOrAfter(LegacyESVersion.V_6_4_0) && (request.preference() != null || request.indexRoutings().length > 0)) {
+        if ((request.preference() != null || request.indexRoutings().length > 0)) {
```
Comment (reta) [inline server/src/main/java/org/opensearch/search/slice/SliceBuilder.java:230]: Super minor, `((` is redundant here, just `(` is enough

**Review by reta — APPROVED** [review]

File: server/src/test/java/org/opensearch/action/search/SearchRequestTests.java [hunk 2]
```diff
@@ -134,25 +132,6 @@ public void testRandomVersionSerialization() throws IOException {
         }
     }
 
-    public void testReadFromPre6_7_0() throws IOException {
-        String msg = "AAEBBWluZGV4AAAAAQACAAAA/////w8AAAAAAAAA/////w8AAAAAAAACAAAAAAABAAMCBAUBAAKABACAAQIAAA==";
-        try (StreamInput in = StreamInput.wrap(Base64.getDecoder().decode(msg))) {
-            in.setVersion(
-                VersionUtils.randomVersionBetween(
-                    random(),
-                    LegacyESVersion.V_6_4_0,
```
Comment (VachaShah) [inline server/src/test/java/org/opensearch/action/search/SearchRequestTests.java:143]: Would this test work for versions between `V_6_5_0` and `V_6_7_0`?

Comment (adnapibar) [reply server/src/test/java/org/opensearch/action/search/SearchRequestTests.java:143]: This is no longer required, as we are removing all 6.x constants eventually in subsequent PRs. See #1674

Comment (VachaShah) [reply server/src/test/java/org/opensearch/action/search/SearchRequestTests.java:143]: Makes sense!

**Review by VachaShah — APPROVED** [review]

File: server/src/main/java/org/opensearch/search/slice/SliceBuilder.java (hunk 1 above)
Comment (adnapibar) [reply server/src/main/java/org/opensearch/search/slice/SliceBuilder.java:230]: will clean up

**Comment by adnapibar** [comment]
Unrelated & un-reproducible failure

```
REPRODUCE WITH: ./gradlew ':server:internalClusterTest' --tests "org.opensearch.indices.stats.IndexStatsIT.testFilterCacheStats" -Dtests.seed=80C55F59F84BA870 -Dtests.security.manager=true -Dtests.jvm.argline="-XX:TieredStopAtLevel=1 -XX:ReservedCodeCacheSize=64m" -Dtests.locale=ar-MA -Dtests.timezone=US/Pacific -Druntime.java=17

org.opensearch.indices.stats.IndexStatsIT > testFilterCacheStats FAILED
    java.lang.AssertionError: shard [index][1] on node [node_s0] has pending operations:
     --> RetentionLeaseBackgroundSyncAction.Request{retentionLeases=RetentionLeases{primaryTerm=1, version=397, leases={peer_recovery/yxVn6WJyS-aXPxJftRrXqQ=RetentionLease{id='peer_recovery/yxVn6WJyS-aXPxJftRrXqQ', retainingSequenceNumber=2, timestamp=1640208027646, source='peer recovery'}}}, shardId=[index][1], timeout=1m, index='index', waitForActiveShards=0}
    	at org.opensearch.index.shard.IndexShardOperationPermits.acquire(IndexShardOperationPermits.java:248)
    	at org.opensearch.index.shard.IndexShard.acquirePrimaryOperationPermit(IndexShard.java:3231)
    	at org.opensearch.action.support.replication.TransportReplicationAction.acquirePrimaryOperationPermit(TransportReplicationAction.java:1117)
    	at org.opensearch.action.support.replication.TransportReplicationAction$AsyncPrimaryAction.doRun(TransportReplicationAction.java:434)
    	at org.opensearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:50)
    	at org.opensearch.action.support.replication.TransportReplicationAction.handlePrimaryRequest(TransportReplicationAction.java:378)
    	at org.opensearch.transport.RequestHandlerRegistry.processMessageReceived(RequestHandlerRegistry.java:98)
    	at org.opensearch.transport.TransportService$8.doRun(TransportService.java:944)
    	at org.opensearch.common.util.concurrent.ThreadContext$ContextPreservingAbstractRunnable.doRun(ThreadContext.java:792)
    	at org.opensearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:50)
    	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
    	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
```

**Comment by adnapibar** [comment]
start gradle check

**Comment by adnapibar** [comment]
More flakiness

```
REPRODUCE WITH: ./gradlew ':qa:remote-clusters:integTest' --tests "org.opensearch.cluster.remote.test.RemoteClustersIT.testHAProxyModeConnectionWorks" -Dtests.seed=2D4D59BC490DD2BB -Dtests.security.manager=true -Dtests.jvm.argline="-XX:TieredStopAtLevel=1 -XX:ReservedCodeCacheSize=64m" -Dtests.locale=sr-CS -Dtests.timezone=Etc/GMT+8 -Druntime.java=17

org.opensearch.cluster.remote.test.RemoteClustersIT > testHAProxyModeConnectionWorks FAILED
    java.lang.AssertionError
        at __randomizedtesting.SeedInfo.seed([2D4D59BC490DD2BB:2A3B8C77728BA4E6]:0)
        at org.junit.Assert.fail(Assert.java:86)
        at org.junit.Assert.assertTrue(Assert.java:41)
        at org.junit.Assert.assertTrue(Assert.java:52)
        at org.opensearch.cluster.remote.test.RemoteClustersIT.testHAProxyModeConnectionWorks(RemoteClustersIT.java:125)

```

**Comment by adnapibar** [comment]
start gradle check

**Comment by adnapibar** [comment]
```
* What went wrong:
Execution failed for task ':distribution:bwc:bugfix:buildBwcLinuxTar'.
> Building 1.2.3 didn't generate expected file /var/CITOOL/workflow/OpenSearch_CI/PR_Checks/Gradle_Check/search/distribution/bwc/bugfix/build/bwc/checkout-1.2/distribution/archives/linux-tar/build/distributions/opensearch-min-1.2.3-SNAPSHOT-linux-x64.tar.gz
```

**Review by andrross — APPROVED** [review]

**Pull request merged**